lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -445,19 +495,45 @@ class UsdPickHandler : public MtohRenderOverride::PickHandlerBase {
 
         // For the USD pick handler pick results are directly returned with USD
         // scene paths, so no need to remove scene index plugin path prefix.
-        const auto& [pickedPath, instanceNdx] = hitPath(pickInput.pickHit);
-        Ufe::Path interpretedPath(registration->interpretRprimPathFn(
-            registration->pluginSceneIndex, pickedPath));
-
-        // Appending a numeric component to the path to identify a point
-        // instance cannot be done on the picked SdfPath, as numeric path
-        // components are not allowed by SdfPath.  Do so here with Ufe::Path,
-        // which has no such restriction.
-        if (instanceNdx >= 0) {
-            interpretedPath = interpretedPath + std::to_string(instanceNdx);
-        }
+        const auto& [pickedUsdPath, instanceNdx] = hitPath(pickInput.pickHit);
+
+        const auto pickedMayaPath = usdPathToUfePath(registration, pickedUsdPath);
+        const auto snMayaPath = (instanceNdx >= 0) ?
+
+            // Point instance: add the instance index to the path.  Appending a
+            // numeric component to the path to identify a point instance
+            // cannot be done on the picked SdfPath, as numeric path components
+            // are not allowed by SdfPath.  Do so here with Ufe::Path, which
+            // has no such restriction.
+            (pickedMayaPath + std::to_string(instanceNdx)) :
+
+            // Not an instance: adjust picked path for selection kind.
+            // As per https://stackoverflow.com/questions/46114214
+            // structured bindings cannot be captured by a lambda in C++ 17,
+            // so pass in pickedUsdPath as a lambda argument.
+            [&pickedMayaPath, &registration](const SdfPath& pickedUsdPath) {
+            auto snKind = GetSelectionKind();
+            if (snKind.IsEmpty()) {
+                return pickedMayaPath;
+            }
+
+            // Get the prim from the stage and path, to access the
+            // UsdModelAPI for the prim.
+            auto proxyShapeObj = registration->dagNode.object();
+            if (proxyShapeObj.isNull()) {
+                TF_FATAL_ERROR("No mayaUsd proxy shape object corresponds to USD pick");
+                return pickedMayaPath;
+            }
+
+            MayaUsdAPI::ProxyStage proxyStage{proxyShapeObj};
+            auto prim = proxyStage.getUsdStage()->GetPrimAtPath(pickedUsdPath);
+            prim = GetPrimOrAncestorWithKind(prim, snKind);
+            const auto usdPath = prim ? prim.GetPath() : pickedUsdPath;
+
+            return usdPathToUfePath(registration, usdPath);
+        }(pickedUsdPath);
```

Comment:
Suggestion : I find the lambda nested in a tertiary a bit hard to read personally, I would propose moving the `(instanceNdx >= 0)` case inside the lambda as well and handle it with an early return like the other cases, so that it's a bit more consistent and `snMayaPath` is directly the result of the lambda.